src/objects.rs

```diff
@@ -1277,9 +1288,81 @@ pub struct ODRule {
     pub network_id: Option<String>,
     pub physical_mode_id: Option<String>,
 }
+impl_id!(ODRuleV1);
 
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[cfg(feature = "fare-v2")]
+pub struct Ticket {
+    pub id: String,
+    pub name: String,
+    pub comment: String,
+}
+#[cfg(feature = "fare-v2")]
 impl_id!(Ticket);
-impl_id!(ODRule);
+
+#[cfg(feature = "fare-v2")]
+impl GetObjectType for Ticket {
+    fn get_object_type() -> ObjectType {
+        ObjectType::Ticket
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[cfg(feature = "fare-v2")]
+pub struct TicketPrice {
+    pub ticket_id: String,
+    pub price: f64, // TODO: how to ensure this value is always positive (u64 on cents? Setter?)
+    pub currency: String, //TODO: evaluate crates 'iso4217' and 'steel-cent' for encoding currency
+    pub validity: ValidityPeriod,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[cfg(feature = "fare-v2")]
+pub struct TicketUse {
+    pub id: String,
+    pub ticket_id: String,
+    pub max_transfers: Option<u64>,
```

Comment:
are you sure the 3 fiels below are optional?

Reply:
So, in the [specification](https://github.com/prhod/navitia/blob/doc_new_fares/documentation/ntfs/ntfs_fare_extension.md#ticket_usestxt-required), they are all marked as **Required**. However, the Description says it can be empty. After some discussion, the specification says **Required** because when it is empty, it still has a meaning (on the contrary of an **Optional** field): for example, `max_transfers` empty means `you can do as many transfers as you like`. I thought `Option<>` was still the good way to model that but we could also do `String` everywhere which can be empty too. What do you think?